common-util/src/main/java/org/sunbird/common/request/BaseRequestValidator.java

```diff
@@ -170,4 +202,64 @@ public void checkMandatoryHeadersPresent(Map<String, String[]> data, String... k
               }
             });
   }
+
+  /**
+   * Method to check unnecessary fields in request
+   *
+   * @param requestMap Request information
+   * @param fields List of not allowed fields
+   */
+  public void checkForFieldsNotAllowed(Map<String, Object> requestMap, List<String> fields) {
+    fields
+        .stream()
+        .forEach(
+            field -> {
+              if (requestMap.containsKey(field)) {
+                throw new ProjectCommonException(
+                    ResponseCode.invalidRequestParameter.getErrorCode(),
+                    ProjectUtil.formatMessage(
+                        ResponseCode.invalidRequestParameter.getErrorMessage(), field),
+                    ResponseCode.CLIENT_ERROR.getResponseCode());
+              }
+            });
+  }
+
+  /**
+   * This method will validate the instance type as List for given fields in request.
+   *
+   * @param requestMap Request information
+   * @param fields List of fields
+   */
+  public void validateListParam(Map<String, Object> requestMap, String... fields) {
+    Arrays.stream(fields)
+        .forEach(
+            field -> {
+              if (requestMap.containsKey(field)
+                  && null != requestMap.get(field)
+                  && !(requestMap.get(field) instanceof List)) {
+                throw new ProjectCommonException(
+                    ResponseCode.dataTypeError.getErrorCode(),
+                    ProjectUtil.formatMessage(
+                        ResponseCode.dataTypeError.getErrorMessage(), field, JsonKey.LIST),
+                    ResponseCode.CLIENT_ERROR.getResponseCode());
+              }
+            });
+  }
+
+  /**
+   * This method will validate the date with format "yyyy-MM-dd".
```

Comment:
Helper method which throws an exception if given date is not in YYYY-MM-DD format.

dob Date of birth